src/ring_theory/integral_closure.lean

```diff
@@ -125,6 +125,18 @@ begin
     aeval_alg_hom_apply, aeval_map_algebra_map, aeval_def, hP.2, _root_.map_zero]
 end
 
+lemma is_integral_map_of_comp_eq_of_is_integral {R S T U : Type*} [comm_ring R] [comm_ring S]
```

Comment:
Can you check if some `comm_ring` can become `comm_semiring` or something similar?

Reply:
I had a got at that but it seems all the `comm_ring` assumptions are necessary

Reply:
That's perfectly fine, it `comm_semiring` works without modification then why not, but I wouldn't spend time on it.